src/wasm/wasm-type.cpp

```diff
@@ -285,6 +289,10 @@ template<typename Info> struct Store {
     typeIDs[info] = id;
     return typename Info::type_t(id);
   }
+
+  void insertWithoutCanonicalizing(std::unique_ptr<Info>&& info) {
```

Comment:
`canonicalize` does not take ownership of anything because it copies the type info object, so I think it's current name describes its intended interface. I got rid of this new method, though.